### DISCUSSION TITLE
Unchecking Purchased Items

### DISCUSSION THREAD

**Discussion by nick-zanetti** [the-collab-lab/tcl-26-smart-shopping-list#38] [Q&A] [answered]
This relates to @anderswift and @SheilaRMoore [PR](https://github.com/the-collab-lab/tcl-26-smart-shopping-list/pull/35) addressing #8 

As the AC is written, I believe the PR follows/fulfills it exactly. I am wondering though, was it TCL's intention to have the item automatically be unchecked 24 hours after purchase? Looking at this line from the Issue:

> Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

Currently, this PR seems to follow that by tying a controlled input to the `recentlyPurchased` state: 

`checked={recentlyPurchased}`

From a UX perspective, it seems a little strange to me. If I check an item off my list, wouldn't I want to manually un check it if I need it again after 24 hours? Currently, the item is unchecked automatically. 

If you comment out this line: `checked={recentlyPurchased}`, the app will behave like I am suggesting, however we do get an error/warning about changing the component from controlled to uncontrolled: 
![Screen Shot 2021-08-05 at 7 24 12 PM](https://user-images.githubusercontent.com/64228185/128433814-62579b08-0d0d-4077-b141-b8ac0e958eb5.png)

Maybe mentors can chime in on this since it involves the wording of the AC?

**Comment by yenly** [comment, marked as answer]
> wouldn't I want to manually un check it if I need it again after 24 hours? Currently, the item is unchecked automatically.

The intention of the "smart" app is to relieve our users from manual actions. It's a safe assumption to make that if you bought bread today, you may want to buy fresh bread again tomorrow morning. For the user to uncheck and then check again, means the user needs to do two actions. From an interactions point of view, we want to use an app to help us do less work than pen & paper.

**Comment by anderswift** [comment]
@nick-zanetti I think this behaviour will make more sense in the context of the completed app, since the list items are going to eventually be ordered based on when the user is expected to buy them again. Right now, it's easy to focus on an individual, unsorted item that is checking and unchecking itself after a kind of arbitrary-seeming 24 hours. But in the completed app, you'll have an automatically-updated list of groceries showing the items you're predicted to need next on top, items you won't need for a while at the bottom, etc. If a recently auto-unchecked item is appearing logically sorted into this list, it will probably be more intuitive to understand what's going on.

**Comment by nick-zanetti** [comment]
This all makes sense, thanks! @yenly @anderswift
